src/main/java/nebula/plugin/metrics/collector/GradleCollector.java

```diff
@@ -171,24 +175,27 @@ public void buildFinished(BuildProfile result) {
      * Conditionally shutdown the dispatcher, because Gradle listener event order appears to be non-deterministic.
      */
     private void shutdownIfComplete() {
+        // only shut down if you have updated build results AND profile information
+        if (!buildProfileComplete.get() || !buildResultComplete.get()) {
```

Comment:
Possibly. There was also an issue (which we saw together) where the build finished before the first batch of JSON was sent up to ES.